src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -189,12 +192,12 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             return;
         }
 
-        private static bool IsForEachIterationVariableDiagnostic(Diagnostic diagnostic, Document document, CancellationToken cancellationToken)
+        private bool IsForEachIterationVariableDiagnostic(Diagnostic diagnostic, Document document, CancellationToken cancellationToken)
         {
             // Do not offer a fix to replace unused foreach iteration variable with discard.
             // User should probably replace it with a for loop based on the collection length.
             var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            return syntaxFacts.IsForEachStatement(diagnostic.Location.FindNode(cancellationToken));
+            return syntaxFacts.IsForEachStatement(UnwrapGlobalStatementIfNecessary(diagnostic.Location.FindNode(cancellationToken)));
```

Comment:
can remove UnwrapGlobalStatementIfNecessary and replace this with `.FindNode(getInnermostNodeForTie: true, cancellationToken);`

Reply:
LOL, this makes all the ceremony with `UnwrapGlobalStatementIfNecessary` actually unnecessary)